fdo.go

```diff
@@ -0,0 +1,264 @@
+package desktop
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"fyne.io/fyne"
+)
+
+var iconExtensions = []string{".png", ".svg", ".xpm"}
+
+//FdoDesktop contains the information from Linux .desktop files
+type FdoDesktop struct {
```

Comment:
This does not seem to be Fdo specific, and we should not really export a namespaced type name.
We could search and replace this as "IconData" type or create an IconData interface that this type privately implements.
This is because we don't want to pass any FDO naming outside of the implementation detail for the type.

Reply:
It is Fdo specific as it is all functions related to fdo spec.  Also this currently has to be exported for use by the window border to get the border icon and that is in a different package from desktop

Reply:
I took a stab at doing an interface, if it sucks I will gladly revert and we can move forward as planned.